GEOS and PROJ bump causing Wave Energy / Coastal Vulnerability tests to fail

I noticed tests failing not related to PR #475 . When looking at the conda env diff, I saw some important updates. 

When updating to Proj=7.2.0 in Conda locally I could reproduce the test failures and I also got the dependent updates:

```
The following packages will be UPDATED:

  cairo                                1.16.0-hd28d34b_1006 --> 1.16.0-hba8bd2f_1007
  expat                                    2.2.9-h33f27b4_2 --> 2.2.10-h39d44d4_0
  gdal                                 3.2.1-py38hb9e1232_0 --> 3.2.1-py38hb9e1232_4
  geos                                     3.8.1-he025d50_0 --> 3.9.0-h39d44d4_0
  geotiff                                  1.6.0-h8884d1a_3 --> 1.6.0-h5d775bf_4
  icu                                       67.1-h33f27b4_0 --> 68.1-h0e60522_0
  libclang                        10.0.1-default_hf44288c_1 --> 11.0.1-default_h5c34c98_1
  libgdal                                  3.2.1-h0d43894_0 --> 3.2.1-hd376e63_4
  libspatialite                            5.0.0-hf693123_0 --> 5.0.1-h6966176_2
  openjpeg                                 2.3.1-h48faf41_3 --> 2.4.0-h48faf41_0
  poppler                                 0.89.0-h5d62644_1 --> 0.89.0-h3772339_5
  proj                                     7.1.1-h7d85306_3 --> 7.2.0-h3e70539_1
  qt                                      5.12.9-hb2cf2c5_0 --> 5.12.9-h5909a2a_3
  shapely                              1.7.1-py38hc96c142_1 --> 1.7.1-py38hc55ee08_2
  tiledb                                   2.1.6-hf84e3da_1 --> 2.2.3-hf84e3da_0
```

For CV:
`test_shore_points_on_multi_polygon` passes
`test_shore_points_on_single_polygon` fails with:
```
>       coastal_vulnerability.prepare_landmass_line_index_and_interpolate_shore_points(
            aoi_path, landmass_path, model_resolution, target_vector_path,
            polygon_pickle, lines_pickle, lines_rtree)

tests\test_coastal_vulnerability.py:963:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
..\venv-space\pyinstaller-41\lib\site-packages\natcap.invest-3.9.0.post185+g9cb3c7d6.d20210208-py3.8-win-amd64.egg\natcap\invest\coastal_vulnerability.py:722: in prepare_landmass_line_index_and_interpolate_shore_points
    unioned_line = shapely.ops.linemerge(lines_in_aoi_list)
..\venv-space\pyinstaller-41\lib\site-packages\shapely\ops.py:120: in linemerge
    result = lgeos.GEOSLineMerge(source._geom)
..\venv-space\pyinstaller-41\lib\site-packages\shapely\geometry\proxy.py:34: in _geom
    self.__geom__, n = self.factory(self.context)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

ob = [[], [], [], []]

    def geos_multilinestring_from_py(ob):
        # ob must be either a MultiLineString, a sequence, or
        # array of sequences or arrays

        if isinstance(ob, MultiLineString):
             return geos_geom_from_py(ob)

        obs = getattr(ob, 'geoms', ob)
        L = len(obs)
        assert L >= 1
        exemplar = obs[0]
        try:
>           N = len(exemplar[0])
E           IndexError: list index out of range

..\venv-space\pyinstaller-41\lib\site-packages\shapely\geometry\multilinestring.py:124: IndexError
```

Alright, so it looks like PROJ 7.2.0 IS causing a Wave Energy Test failure:

```python
================================== FAILURES ===================================
______ WaveEnergyUnitTests.test_pixel_size_based_on_coordinate_transform ______

...

     # Call the function to test
     result = wave_energy._pixel_size_based_on_coordinate_transform(
             raster_path, coord_trans, point)
   
     expected_res = (5553.933063, -1187.370813)
  
     # Compare
     for res, exp in zip(result, expected_res):
          self.assertAlmostEqual(res, exp, places=5)
           AssertionError: 5553.932270178571 != 5553.933063 within 5 places (0.0007928214290586766 difference)
```

--- a/Makefile
+++ b/Makefile
@@ -6,7 +6,7 @@ GIT_SAMPLE_DATA_REPO_REV    := b7a51f189315e08484b5ba997a5c1de88ab7f06d
 
 GIT_TEST_DATA_REPO          := https://bitbucket.org/natcap/invest-test-data.git
 GIT_TEST_DATA_REPO_PATH     := $(DATA_DIR)/invest-test-data
-GIT_TEST_DATA_REPO_REV      := 69ea5fc56906c029bc745e5531b7a4d4e4c13237
+GIT_TEST_DATA_REPO_REV      := 0057a412104fbf97d1777bfffa3ad725485b9e02
 
 GIT_UG_REPO                  := https://github.com/natcap/invest.users-guide
 GIT_UG_REPO_PATH             := doc/users-guide

--- a/src/natcap/invest/utils.py
+++ b/src/natcap/invest/utils.py
@@ -773,11 +773,18 @@ def _assert_vectors_equal(
             actual_geom_wkt = actual_geom.ExportToWkt()
             expected_geom_shapely = loads(expected_geom_wkt)
             actual_geom_shapely = loads(actual_geom_wkt)
-            if not expected_geom_shapely.almost_equals(actual_geom_shapely):
-                raise AssertionError(
-                    "Vector geometry assertion fail. \n"
-                    f"Expected geometry: {expected_geom_wkt}. \n"
-                    f"Actual geometry: {actual_geom_wkt}. ")
+            # Try comparing geoms exactly equal allowing for different
+            # geometry ordering
+            geoms_equal = expected_geom_shapely.equals(actual_geom_shapely)
+            if not geoms_equal:
+                # Try almost_equal allowing for precision differences
+                geoms_almost_eq = expected_geom_shapely.almost_equals(
+                    actual_geom_shapely)
+                if not geoms_almost_qual:
+                    raise AssertionError(
+                        "Vector geometry assertion fail. \n"
+                        f"Expected geometry: {expected_geom_wkt}. \n"
+                        f"Actual geometry: {actual_geom_wkt}. ")
 
             expected_feature = None
             actual_feature = None

--- a/src/natcap/invest/utils.py
+++ b/src/natcap/invest/utils.py
@@ -780,7 +780,7 @@ def _assert_vectors_equal(
                 # Try almost_equal allowing for precision differences
                 geoms_almost_eq = expected_geom_shapely.almost_equals(
                     actual_geom_shapely)
-                if not geoms_almost_qual:
+                if not geoms_almost_eq:
                     raise AssertionError(
                         "Vector geometry assertion fail. \n"
                         f"Expected geometry: {expected_geom_wkt}. \n"

--- a/tests/test_wave_energy.py
+++ b/tests/test_wave_energy.py
@@ -70,7 +70,6 @@ class WaveEnergyUnitTests(unittest.TestCase):
         """Overriding tearDown function to remove temporary directory."""
         shutil.rmtree(self.workspace_dir)
 
-    @unittest.skip("Skipping for PROJ 7.2.0 update.")
     def test_pixel_size_based_on_coordinate_transform(self):
         """WaveEnergy: test '_pixel_size_based_on_coordinate_transform' fn."""
         from natcap.invest import wave_energy
@@ -117,7 +116,7 @@ class WaveEnergyUnitTests(unittest.TestCase):
 
         # Compare
         for res, exp in zip(result, expected_res):
-            self.assertAlmostEqual(res, exp, places=5)
+            self.assertAlmostEqual(res, exp, places=2)
 
     def test_count_pixels_groups(self):
         """WaveEnergy: testing '_count_pixels_groups' function."""
@@ -170,25 +169,62 @@ class WaveEnergyUnitTests(unittest.TestCase):
         for res, exp_res in zip(result_id, expected_result_id):
             self.assertEqual(res, exp_res)
 
-    @unittest.skip("Skipping for GEOS 3.9.0 bug.")
     def test_clip_vector_by_vector_polygons(self):
         """WaveEnergy: testing clipping polygons from polygons."""
         from natcap.invest import wave_energy
+        from natcap.invest.utils import _assert_vectors_equal
 
-        aoi_path = os.path.join(REGRESSION_DATA, 'aoi_proj_to_extract.shp')
-        extract_path = os.path.join(
-            SAMPLE_DATA, 'WaveData', 'Global_extract.shp')
+        projection_wkt = osr.SRS_WKT_WGS84_LAT_LONG
+        origin = (-62.00, 44.00)
+        pos_x = origin[0]
+        pos_y = origin[1]
 
-        result_path = os.path.join(self.workspace_dir, 'aoi_proj_clipped.shp')
-        target_projection = pygeoprocessing.get_vector_info(
-            extract_path)['projection_wkt']
+        fields_aoi = {'id': ogr.OFTInteger}
+        attrs_aoi = [{'id': 1}]
+        # Create polygon for the aoi
+        aoi_polygon = [
+            Polygon([(pos_x, pos_y), (pos_x + 2, pos_y),
+                     (pos_x + 2, pos_y - 2), (pos_x, pos_y - 2),
+                     (pos_x, pos_y)])
+        ]
+
+        aoi_path = os.path.join(self.workspace_dir, 'aoi.shp')
+        # Create the polygon shapefile
+        pygeoprocessing.shapely_geometry_to_vector(
+            aoi_polygon, aoi_path, projection_wkt, 'ESRI Shapefile',
+            fields=fields_aoi, attribute_list=attrs_aoi)
+
+        fields_data = {'id': ogr.OFTInteger, 'myattr': ogr.OFTString}
+        attrs_data = [{'id': 1, 'myattr': 'hello'}]
+        # Create polygon to clip with the aoi
+        data_polygon = [
+            Polygon([(pos_x - 2, pos_y + 2), (pos_x + 6, pos_y - 2),
+                     (pos_x + 6, pos_y - 4), (pos_x - 2, pos_y - 6),
+                     (pos_x - 2, pos_y + 2)])
+        ]
+
+        data_path = os.path.join(self.workspace_dir, 'data.shp')
+        # Create the polygon shapefile
+        pygeoprocessing.shapely_geometry_to_vector(
+            data_polygon, data_path, projection_wkt, 'ESRI Shapefile',
+            fields=fields_data, attribute_list=attrs_data)
+
+        result_path = os.path.join(self.workspace_dir, 'aoi_clipped.shp')
         wave_energy._clip_vector_by_vector(
-            aoi_path, extract_path, result_path, target_projection,
+            data_path, aoi_path, result_path, projection_wkt,
             self.workspace_dir)
 
-        expected_path = os.path.join(REGRESSION_DATA, 'aoi_proj_clipped.shp')
-        WaveEnergyRegressionTests._assert_point_vectors_equal(
-            result_path, expected_path)
+        fields_expected = {'id': ogr.OFTInteger, 'myattr': ogr.OFTString}
+        attrs_expected = [{'id': 1, 'myattr': 'hello'}]
+        # Create polygon to clip with the aoi
+        expected_polygon = aoi_polygon
+        expected_path = os.path.join(self.workspace_dir, 'expected.shp')
+        # Create the polygon shapefile
+        pygeoprocessing.shapely_geometry_to_vector(
+            expected_polygon, expected_path, projection_wkt, 'ESRI Shapefile',
+            fields=fields_expected, attribute_list=attrs_expected)
+
+        _assert_vectors_equal(expected_path, result_path)
 
     def test_clip_vector_by_vector_points(self):
         """WaveEnergy: testing clipping points from polygons."""

